tests/integration/security/rbac/util/rbac_util.go

```diff
@@ -56,37 +55,32 @@ func CheckRBACRequest(tc TestCase) error {
 		return fmt.Errorf("connection error with %v", err)
 	}
 
-	if tc.ExpectAllowed {
-		if !(len(resp) > 0 && resp[0].Code == connection.AllowHTTPRespCode) {
-			return fmt.Errorf("%s to %s:%d%s using %s: expected allow, actually deny",
-				req.From.Name(), req.To.Name(), req.Port, req.Path, req.Protocol)
-		}
-	} else {
+	if err != nil {
```

Comment:
I suggest to change to the following flow:
```
if tc.ExpectedRespCode == connection.DenyHTTPRespCode {
  if req.Port == connection.TCPPort {
    if !strings.Contains(err.Error(), "EOF") {
      t.Errorf("failed, expect EOF for tcp but got: ...");
    }
  } else {
    if !(len(resp) > 0 && resp[0].Code == connection.DenyHTTPRespCode) {
      t.Errorf("failed, expect 403 for http but got: ...");
    }
  }
} else if tc.ExpectedRespCode == connection.AllowHTTPRespCode {
  if !(len(resp) > 0 && resp[0].Code == connection.AllowHTTPRespCode) {
    t.Errorf("failed, expect 200 response code but got: ...");
  }
} else {
  t.Errorf("unsupported expected resp code: %s", tc.ExpectedRespCode)
}
```

The point is to always based on the expectation `tc.ExpectedRespCode` in the outer if-else statement.

Reply:
This will segfault if `tc.ExpectedRespCode == connection.DenyHTTPRespCode`, `req.Port == connection.TCPPort`, and `err` is nil though? I think the current implementation is shorter and not harder to understand. Wdyt?

Reply:
That sounds reasonable. Will do. Thanks.

Reply:
Actually, let's switch back to our old model, using `expectAllowed`. We should abstract the test writers away from thinking about http resp code. RBAC should either be deny or allow. If there are multiple types of denies or allows, we can enhance this `CheckRBACRequest` function later using the resp body as you (@yangminzhu) mentioned. Using Http resp code as the expectation from the test writer doesn't make sense in RBAC.

/cc @lei-tang for visibility. FWIW, `CheckConnection` are still using `ExpectedSuccess`, rather than http resp code.